src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,75 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useState } from 'react';
+// eslint-disable-next-line import/extensions
+import { FilterCriteria, Status } from '../../types/otherTypes';
+import React from 'react';
+type Props = {
+  onFilterChange: (filterCriteria: FilterCriteria) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ onFilterChange }) => {
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [searchInput, setSearchInput] = useState('');
+
```

Comment:
You store these values in the parent component in `filter` state. Get rid of them here, you have to have one source of truth